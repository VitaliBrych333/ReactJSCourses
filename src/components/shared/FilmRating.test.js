import React from 'react';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';
import Rating from './FilmRating';

describe('<Rating/>', () => {
  const initialState = {
    movieReducer: {
      filmId: {
        data: {
          vote_average: 1,
          title: 'test',
          tagline: 'test',
        },
      },
    },
  };
  const mockStore = configureStore();
  let store;

  beforeEach(() => {
    container = document.createElement('div');
    document.body.appendChild(container);
    store = mockStore(initialState);
  });

  afterEach(() => {
    document.body.removeChild(container);
    container = null;
  });

  it('should equals to snapshot of Rating', () => {
    const renderedValue = renderer
      .create(
        <Provider store={store}>
          <Rating />
        </Provider>
      )
      .toJSON();
    expect(renderedValue).toMatchSnapshot();
  });
});
