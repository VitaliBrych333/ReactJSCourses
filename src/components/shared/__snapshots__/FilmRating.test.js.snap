// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Rating/> should equals to snapshot of Rating 1`] = `
<div
  className="sc-bdVaJa jGiXbe"
>
  <h3>
    test
     
    <span>
      test
    </span>
  </h3>
  <div>
    1
  </div>
</div>
`;
